File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> st;
        
        for(int i:nums) st.insert(i);
        
        int res=0;
        
        for(int i:nums) {
            if(st.count(i-1)>0) continue;
            else {
                int curr=0, val=i;
                while(st.count(val)>0) {
                    curr++;
                    val++;
                }
                res=max(res, curr);
            }
        }
        
        return res;
    }
};

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        numbers=set()
        
        for i in nums:
            numbers.add(i)
        
        res=0
        for i in nums:
            if i-1 not in numbers:
                curr=1
                while(i+curr in numbers):
                    curr=curr+1
                if res<curr:
                    res=curr
        return res